### ISSUE TITLE
Evaluating error

### ISSUE THREAD

**Issue by TommeyChang** [chihyaoma/regretful-agent#4]
Evaluating on val_seen env ...
Epoch: [1][1/16]	Time 1.776 (1.776)	Loss inf (inf)	
Epoch: [1][2/16]	Time 1.733 (1.754)	Loss inf (inf)	
Epoch: [1][3/16]	Time 1.885 (1.798)	Loss inf (inf)	
Epoch: [1][4/16]	Time 1.867 (1.815)	Loss inf (inf)	
Epoch: [1][5/16]	Time 1.919 (1.836)	Loss inf (inf)	
Epoch: [1][6/16]	Time 1.874 (1.842)	Loss inf (inf)	
Epoch: [1][7/16]	Time 1.779 (1.833)	Loss inf (inf)	
Epoch: [1][8/16]	Time 1.958 (1.849)	Loss inf (inf)	
Epoch: [1][9/16]	Time 1.626 (1.824)	Loss inf (inf)	
Warning: NaN or Inf found in input tensor.
Epoch: [1][10/16]	Time 1.899 (1.832)	Loss inf (inf)	
Epoch: [1][11/16]	Time 1.697 (1.819)	Loss inf (inf)	
Epoch: [1][12/16]	Time 1.781 (1.816)	Loss inf (inf)	
Epoch: [1][13/16]	Time 1.938 (1.826)	Loss inf (inf)	
Epoch: [1][14/16]	Time 1.610 (1.810)	Loss inf (inf)	
Epoch: [1][15/16]	Time 1.358 (1.780)	Loss inf (inf)	
Epoch: [1][16/16]	Time 1.543 (1.765)	Loss inf (inf)

**Comment by chihyaoma** [comment]
Hi, 

This is perfectly normal for the **regretful agent** since we have `prevent_oscillation` enabled. What this means is that, we block a direction that will lead to oscillation, i.e., a vector for that particular direction is being masked.

However, if that direction is actually the ground-truth the agent should pick (during inference), the computed loss will be effective `inf`. This will only happen during inference though (it won't affect your training).  



However, I haven't seen the warning message before. Perhaps it's a warning from new PyTorch version?

**Comment by TommeyChang** [comment]
But it seem no warning while training.
